kb/deadline-exceeded-troubleshooting.md

```diff
@@ -0,0 +1,99 @@
+---
+slug: deadline-exceeded-troubleshooting
+title: DeadlineExceeded error troubleshooting
+tags:
+  - kb-article
+date: 2022-11-08T00:00:00Z
+---
+
+All requests made to the [Temporal Cluster](/concepts/what-is-a-temporal-cluster) are [gRPC requests](https://grpc.io/docs/what-is-grpc/core-concepts/#deadlines).
+Sometimes, when these requests can't be completed, you'll see this particular error message: `Context: deadline exceeded`.
+Network interruptions, short timeouts, server overload, and Query errors are some of the reasons that can cause this error.
+
+The following sections discuss the nature of this error and how to troubleshoot it.
+
+#### Check your logs
+
+:::note
+
+Cloud users cannot access some of the logs needed to diagnose the source of the error.
+
+If you're using Temporal Cloud, check your Workflow History for errors, look for latencies and errors in your Cloud metrics, and create a [support ticket](https://support.temporal.io/) with Temporal providing these details.
+
+:::
+
+```
+{“level”:“error”,“ts”:“2022-03-21T19:32:42.312Z”,“msg”:“unavailable error”,“service”:“frontend”,“error”:“unable to get temporal-sys-add-search-attributes-workflow workflow state: context deadline exceeded”,“logging-call-at”:“adminHandler.go:1163”,“stacktrace”:“go.temporal.io/server/common/log.
+```
+
+Service logs can show which parts of the Cluster aren't working.
+
+Verify that the Frontend Service is connected by opening the Web UI in your browser.
+
+Check that the following are up and running with `tctl cluster` commands:
+
+- [History Service](/clusters#history-service)
+- [Persistence](/clusters#persistence) database
+- [Advanced Visibility](/visibility#advanced-visibility) database
+
+Logs can also be used to find [Query](/workflows#queries) errors.
+
+#### Check your Cluster metrics
+
+Look for high latencies, short timeouts, and other abnormal [Cluster metrics](/references/cluster-metrics).
+If the metrics come from a specific service (such as History Service), check that service's connectivity and restart if needed.
+
+#### Check Workflow logic
+
+Check your [Client and Worker configuration](/references/configuration) files for missing or invalid values in:
+
+- Server names
```

Comment:
i dont think deadline exceeded would happen in case client / worker is connecting to an invalid target. in this case would probably get connection refused errors

same for network or host address 

for timeouts in this list, im not sure how to check missing timeouts honestly so not sure what to check for or change